src/uci_handler.h

```diff
@@ -107,7 +107,7 @@ class UciHandler {
                 const auto move = parsing::moveFromString(s_board, moveSv.value_or(sv));
 
                 if (move.has_value()) {
-                    s_board = engine::performMove(s_board, move.value(), hash);
+                    s_board = s_board.player == PlayerWhite ? engine::performMove<PlayerWhite>(s_board, move.value(), hash) : engine::performMove<PlayerBlack>(s_board, move.value(), hash);
```

Comment:
Helper can be used here
```suggestion
                    s_board = engine::performMove(s_board, move.value(), hash);
```